app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PaymentsController < ApplicationController
+
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :find_invoice
+
+  def index
+    @payments = @invoice.payments
+  end
+
+  def new
+    @payment = Payment.new
+  end
+
+  def edit
+    @payment = Payment.find(params[:id])
+  end
+
+  def create
+    @payment = @invoice.payments.build(payment_params)
+    @payment.date_of_payment = format_date_locale(params[:payment][:date_of_payment])
+    if @payment.save
+      begin
+        ClientMailer.send_email_payment(@invoice, current_user).deliver if params[:email].present?
+        flash[:success] = 'Payment saved successfully.'
+        redirect_to invoice_payments_path
+      rescue
+        flash[:error] = "Problem while sending email."
+        render action: 'new'
+      end
+    else
+      flash[:error] = "Problem while saving payment details. #{add_flash_messages(@payment)}"
+      render action: 'new'
+    end
+  end
+
+
+  def update
+    params[:payment][:date_of_payment] = format_date_locale(params[:payment][:date_of_payment])
```

Comment:
As suggested earlier replace params with formatted dates and that should happen in before_filter